src/uct/tcp/tcp_listener.c

```diff
@@ -116,11 +115,13 @@ UCS_CLASS_INIT_FUNC(uct_tcp_listener_t, uct_cm_h cm,
         goto err_close_socket;
     }
 
-    ports_range.first = port;
-    ports_range.last  = port;
+    is_port_zero = (port == 0);
+
+    do {
+        status = ucs_socket_server_init(saddr, socklen, backlog, is_port_zero,
+                                        &self->listen_fd);
+    } while (is_port_zero && (status == EADDRINUSE));
```

Comment:
why do we need to retry? if user requested random port it should work same as with regular TCP, e.g it could fail

Reply:
previously we kept trying for port=0 so i kept this behaviour. 

so we shouldn't try more port values if first one fails if port is 0?
from tcp_iface as well?

Reply:
for iface - yes, for listener - no

Reply:
because tcp_iface isn't coming from the user?

Reply:
yes